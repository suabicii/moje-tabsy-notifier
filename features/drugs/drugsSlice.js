import {createSlice} from "@reduxjs/toolkit";
import {ajaxCall} from "../../utils/ajaxCall";


const drugsSlice = createSlice({
    name: 'drugs',
    initialState: [],
    reducers: {
        setDrugs: (state, action) => action.payload
    }
});

const {actions, reducer} = drugsSlice;

export const {setDrugs} = actions;

export const fetchDrugs = token => async dispatch => {
    await ajaxCall('get', `drug-notify/${token}`)
        .then(data => {
            dispatch(setDrugs(data));
        })
        .catch(err => {
            console.log(err);
        });
};

export default reducer;