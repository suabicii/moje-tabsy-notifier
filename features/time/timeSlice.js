import {createSlice} from "@reduxjs/toolkit";

const timeSlice = createSlice({
    name: 'time',
    initialState: {},
    reducers: {
        setCurrentTime: (state, action) => ({...state, now: action.payload}),
        setTomorrowTime: (state, action) => ({...state, tomorrow: action.payload})
    }
});

const {actions, reducer} = timeSlice;

export const {setCurrentTime, setTomorrowTime} = actions;

export default reducer;