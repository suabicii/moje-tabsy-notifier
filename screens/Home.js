import React, {useEffect, useState} from "react";
import {Text, TouchableOpacity, View} from "react-native";
import {APP_ID, APP_TOKEN} from "@env";

function Home({navigation, route}) {
    const {userId, logged} = route.params;
    const [isLogged, setIsLogged] = useState(logged);

    // block going back until user pushed logout button
    navigation.addListener('beforeRemove', e => {
        if (!isLogged) {
            navigation.dispatch(e.data.action); // unblock going back action
            return;
        }
        e.preventDefault();
    });

    useEffect(() => {
        if (!isLogged) {
            navigation.navigate('Login');
        }
    }, [isLogged]);

    // useEffect(() => {
    //     async function sendNotificationAfterLogin() {
    //         await fetch('https://app.nativenotify.com/api/indie/notification', {
    //             method: 'POST',
    //             headers: {
    //                 'Accept': 'application/json',
    //                 'Content-Type': 'application/json'
    //             },
    //             body: JSON.stringify({
    //                 subID: userId,
    //                 appId: APP_ID,
    //                 appToken: APP_TOKEN,
    //                 title: 'Login success',
    //                 message: 'Pomyślnie zalogowano'
    //             })
    //         });
    //     }
    //     sendNotificationAfterLogin();
    // }, []);

    return (
        <View style={{flex: 1, justifyContent: "center"}}>
            <Text style={{
                fontSize: 32,
                fontWeight: "500",
                textAlign: "center"
            }}>
                Monitorowanie rozpoczęte 👁
            </Text>
            <Text style={{
                fontSize: 24,
                marginTop: 25,
                paddingHorizontal: 5,
                textAlign: "center"
            }}>
                Od teraz będziesz otrzymywał/-a powiadomienia 🔔,
                które będą Ci przypominać o zażyciu leków/suplementów 💊
                lub kończących się zapasach
            </Text>
            <Text style={{
                color: "#525252",
                fontSize: 22,
                marginTop: 10,
                paddingHorizontal: 5,
                textAlign: "center"
            }}>
                W godzinie zażycia medykamentu pojawi się przycisk,
                dzięki któremu potwierdzisz, że właśnie został
                przez Ciebie przyjęty ✅
            </Text>
            <TouchableOpacity
                testID="logoutButton"
                onPress={() => setIsLogged(false)}
                style={{
                    backgroundColor: "#f38c4c",
                    borderRadius: 24,
                    height: 50,
                    justifyContent: "center",
                    marginTop: 20,
                    marginHorizontal: 15
                }}
            >
                <Text style={{
                    color: "#fff",
                    fontSize: 24,
                    fontWeight: "300",
                    textAlign: "center"
                }}>
                    Wyloguj się
                </Text>
            </TouchableOpacity>
        </View>
    );
}

export default Home;