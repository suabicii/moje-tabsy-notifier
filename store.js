import {configureStore} from "@reduxjs/toolkit";
import drugsSlice from "./features/drugs/drugsSlice";
import timeSlice from "./features/time/timeSlice";
import drugsTakenSlice from "./features/drugsTaken/drugsTakenSlice";
import expandedAccordionsSlice from "./features/detailsExpanded/expandedAccordionsSlice";
import activeSessionSlice from "./features/activeSession/activeSessionSlice";

export default configureStore({
    reducer: {
        activeSession: activeSessionSlice,
        drugs: drugsSlice,
        drugsTaken: drugsTakenSlice,
        expandedAccordions: expandedAccordionsSlice,
        time: timeSlice
    }
});