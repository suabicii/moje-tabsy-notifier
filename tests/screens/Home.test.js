import React from "react";
import renderer from "react-test-renderer";
import Home from "../../screens/Home";
import {act, render, screen, fireEvent, waitFor} from "@testing-library/react-native";
import {BackHandler} from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";
import MockDate from "mockdate";
import dayjs from "dayjs";
import {Provider} from "react-redux";
import store from "../../store";
import {drugList} from "../fixtures/drugList";
import {fetchDrugs} from "../../features/drugs/drugsSlice";
import {setCurrentTime} from "../../features/time/timeSlice";
import sendNotification from "../../utils/notifier";

let currentTime;
const mockGetHeaders = {get: args => 'application/json'}
beforeAll(() => {
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({
        headers: mockGetHeaders, // Avoid: "TypeError: Cannot read properties of undefined (reading 'get')"
        json: () => Promise.resolve({
            status: 200,
            message: 'successfully logged out'
        })
    }));

    MockDate.set('2020-01-01');
    currentTime = dayjs();

    store.dispatch(setCurrentTime(JSON.stringify(currentTime)));
});

afterEach(() => {
    MockDate.reset();
});

afterAll(() => {
    jest.clearAllMocks();
    delete global.fetch;
});

const addListener = jest.fn();
const navigate = jest.fn();
const dispatch = jest.fn();

const userId = 'john@doe.com';

function WrappedComponent({navigateCustom}) {
    const navigateLocal = navigateCustom || navigate;

    return (
        <Provider store={store}>
            <Home
                route={{
                    params: {logged: true, userId}
                }}
                navigation={{
                    navigate: navigateLocal,
                    addListener,
                    dispatch
                }}
            />
        </Provider>
    );
}

it('should correctly render Home screen', async () => {
    const tree = renderer.create(<WrappedComponent/>).toJSON();
    await waitFor(() => {
        expect(tree).toMatchSnapshot();
    });
});

it('should go back to Login screen after pressing logout button', async () => {
    render(<WrappedComponent/>);

    await act(async () => {
        fireEvent.press(screen.getByTestId('btn-pill'));
    });

    await waitFor(() => {
        expect(navigate).toBeCalled();
    });
});

it('should stay in Home screen if logout request failed', async () => {
    const navigate = jest.fn();
    render(<WrappedComponent navigateCustom={navigate}/>);
    fetch.mockRejectedValueOnce(new Error('Something went wrong'));

    await act(async () => {
        fireEvent.press(screen.getByTestId('btn-pill'));
    });

    expect(navigate).toHaveBeenCalledTimes(0);
});

it('should block going back if user is logged in', async () => {
    const navigate = jest.fn();
    render(<WrappedComponent navigateCustom={navigate}/>);

    BackHandler.mockPressBack();

    await waitFor(() => {
        expect(navigate).toBeCalledTimes(0);
    });
});

it('should show modal with welcome message on first screen load after login', async () => {
    render(<WrappedComponent/>);

    await waitFor(() => {
        expect(screen.getByTestId('welcomeModal')).toBeTruthy();
    });
});

it('should not show modal with welcome message after login if AsyncStorage has information about it', async () => {
    await AsyncStorage.setItem('welcome_msg_disable', 'true');
    render(<WrappedComponent/>);

    await waitFor(() => {
        expect(screen.queryByTestId('welcomeModal')).toBeNull();
    });
});

it('should reload Drug component with drug list after pressing refresh button', async () => {
    jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve(drugList)
    }));
    store.dispatch(fetchDrugs('mock_token'));
    render(<WrappedComponent/>);

    await act(() => {
        fireEvent.press(screen.getByTestId('refreshBtn'));
    });

    await waitFor(() => {
        expect(fetch).toBeCalled();
    });
});

it('should save tomorrow time to local storage', async () => {
    render(<WrappedComponent/>);

    const tomorrowTime = dayjs().add(1, 'd').startOf('d');
    const tomorrowTimeJSON = JSON.stringify(tomorrowTime);

    await waitFor(() => {
        expect(AsyncStorage.setItem).toBeCalledWith('tomorrow_time', tomorrowTimeJSON);
    });
});

it('should remove tomorrow time and save new value in local storage if current time is same or after tomorrow time', async () => {
    const tomorrowTime = dayjs().subtract(1, 'd');
    const tomorrowTimeJSON = JSON.stringify(tomorrowTime);

    await AsyncStorage.setItem('tomorrow_time', tomorrowTimeJSON);

    render(<WrappedComponent/>);

    const tomorrowTimeNewValue = dayjs().add(1, 'd').startOf('d');
    const tomorrowTimeNewValueJSON = JSON.stringify(tomorrowTimeNewValue);

    await waitFor(() => {
        expect(AsyncStorage.setItem).toBeCalledWith('tomorrow_time', tomorrowTimeNewValueJSON);
    });
});

describe('Queue/send notifications', () => {
    beforeAll(() => {
        jest.spyOn(global, 'fetch').mockImplementation(() => Promise.resolve({
            headers: mockGetHeaders,
            json: () => Promise.resolve(drugList)
        }));

        const notifier = require('../../utils/notifier');
        jest.spyOn(notifier, 'default');
    });

    it('should queue notifications', async () => {
        store.dispatch(fetchDrugs('mock_token'));
        render(<WrappedComponent/>);

        await waitFor(() => {
            expect(store.getState().notificationsQueue.length).toBeGreaterThan(0);
        });
    });

    it('should send notification at the appropriate time', async () => {
        const {dosing, name, unit} = drugList[0];
        const dosingTime = dayjs().hour(7).minute(2);
        store.dispatch(setCurrentTime(JSON.stringify(dosingTime)));
        store.dispatch(fetchDrugs('mock_token'));

        render(<WrappedComponent/>);

        await waitFor(() => {
            expect(sendNotification).toBeCalledWith(name, dosing, unit, userId);
        });
    });

    it('should send proper amount of notifications', async () => {
        const dosingTime = dayjs().hour(12).minute(2);
        store.dispatch(setCurrentTime(JSON.stringify(dosingTime)));
        store.dispatch(fetchDrugs('mock_token'));

        render(<WrappedComponent/>);

        /**
         * 3 because:
         * Xanax: 7:00,
         * Witamina C: 12:00
         * Metanabol: 12:00
         * */

        await waitFor(() => {
            expect(sendNotification).toBeCalledTimes(3);
        });
    });
});