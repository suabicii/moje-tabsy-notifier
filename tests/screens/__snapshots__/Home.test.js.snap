// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Home screen 1`] = `
<RCTScrollView>
  <View>
    <Text
      style={
        {
          "fontSize": 32,
          "fontWeight": "500",
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Monitorowanie rozpoczƒôte üëÅ
    </Text>
    <View
      collapsable={false}
      style={
        {
          "alignContent": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "rgba(0, 0, 0, 0.12)",
          "borderRadius": 4,
          "elevation": 1,
          "margin": 10,
          "marginBottom": 20,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 0.5,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 0.75,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 16,
              },
              {
                "minHeight": 50,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "500",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontSize": 20,
                        "letterSpacing": 0.15,
                        "lineHeight": 30,
                        "marginVertical": 2,
                      },
                      [
                        {
                          "minHeight": 30,
                          "paddingRight": 16,
                        },
                        {
                          "marginBottom": 2,
                        },
                        {
                          "color": "#252525",
                          "textAlign": "center",
                        },
                      ],
                    ],
                  ],
                ]
              }
            >
              W najbli≈ºszym czasie muszƒô za≈ºyƒá:
            </Text>
          </View>
          <View />
        </View>
        <View
          style={
            [
              {
                "paddingHorizontal": 16,
              },
              {
                "paddingBottom": 16,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityRole="progressbar"
            accessibilityState={
              {
                "busy": true,
              }
            }
            accessible={true}
            style={
              [
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                {
                  "marginBottom": 10,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "height": 48,
                  "opacity": 1,
                  "width": 48,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 48,
                      "transform": [
                        {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 48,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 24,
                        "overflow": "hidden",
                        "width": 48,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "height": 48,
                          "transform": [
                            {
                              "translateY": 0,
                            },
                            {
                              "rotate": "-165deg",
                            },
                          ],
                          "width": 48,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "height": 24,
                            "overflow": "hidden",
                            "width": 48,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            {
                              "borderColor": "#80d8ff",
                              "borderRadius": 24,
                              "borderWidth": 4.8,
                              "height": 48,
                              "width": 48,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  {
                    "alignItems": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 48,
                      "transform": [
                        {
                          "rotate": "45deg",
                        },
                      ],
                      "width": 48,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 24,
                        "overflow": "hidden",
                        "top": 24,
                        "width": 48,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "height": 48,
                          "transform": [
                            {
                              "translateY": -24,
                            },
                            {
                              "rotate": "345deg",
                            },
                          ],
                          "width": 48,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "height": 24,
                            "overflow": "hidden",
                            "width": 48,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            {
                              "borderColor": "#80d8ff",
                              "borderRadius": 24,
                              "borderWidth": 4.8,
                              "height": 48,
                              "width": 48,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "#78c2ad",
                "borderColor": "transparent",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 2,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 0.75,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 4,
                  },
                ]
              }
              testID="refreshBtn"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    {
                      "marginLeft": 12,
                      "marginRight": -4,
                    }
                  }
                >
                  <Text />
                </View>
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      {
                        "textAlign": "left",
                      },
                      [
                        {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        {
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Od≈õwie≈º
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        testID="welcomeModal"
        visible={true}
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 22,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderRadius": 20,
                "elevation": 5,
                "margin": 20,
                "padding": 35,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 22,
                  "marginBottom": 15,
                  "textAlign": "center",
                }
              }
            >
              Od teraz bƒôdziesz otrzymywa≈Ç/-a powiadomienia üîî, kt√≥re bƒôdƒÖ Ci przypominaƒá o za≈ºyciu lek√≥w/suplement√≥w üíä lub ko≈ÑczƒÖcych siƒô zapasach
            </Text>
            <Text
              style={
                [
                  {
                    "fontSize": 22,
                    "marginBottom": 15,
                    "textAlign": "center",
                  },
                  {
                    "color": "#525252",
                    "fontSize": 20,
                  },
                ]
              }
            >
              W godzinie za≈ºycia medykamentu pojawi siƒô przycisk, dziƒôki kt√≥remu potwierdzisz, ≈ºe w≈Ça≈õnie zosta≈Ç przez Ciebie przyjƒôty ‚úÖ
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              fillColor="#78c2ad"
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  {
                    "marginBottom": 15,
                  },
                ]
              }
              testID="welcomeMsgDisableCheckbox"
              text="Nie pokazuj mi tego wiƒôcej"
              textStyle={
                {
                  "textDecorationLine": "none",
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderRadius": 12.5,
                    "height": 25,
                    "justifyContent": "center",
                    "transform": [
                      {
                        "scale": 1,
                      },
                    ],
                    "width": 25,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "borderColor": "#78c2ad",
                        "borderRadius": 12.5,
                        "borderWidth": 1,
                        "height": 25,
                        "justifyContent": "center",
                        "width": 25,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
              <View
                style={
                  [
                    {
                      "marginLeft": 16,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#757575",
                        "fontSize": 16,
                        "textDecorationLine": "none",
                      },
                      {
                        "textDecorationLine": "none",
                      },
                    ]
                  }
                >
                  Nie pokazuj mi tego wiƒôcej
                </Text>
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "borderRadius": 25,
                    "elevation": 2,
                    "padding": 15,
                  },
                  {
                    "backgroundColor": "#2196F3",
                  },
                ]
              }
            >
              <Text
                style={
                  {
                    "color": "white",
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                OK
              </Text>
            </View>
          </View>
        </View>
      </Modal>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#f38c4c",
          "borderRadius": 24,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 20,
          "marginHorizontal": 15,
          "opacity": 1,
        }
      }
      testID="btn-pill"
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 24,
            "fontWeight": "300",
            "textAlign": "center",
          }
        }
      >
        Wyloguj siƒô
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
